processor/resourcedetectionprocessor/internal/azure/azure.go

```diff
@@ -0,0 +1,64 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package azure
+
+import (
+	"context"
+	"fmt"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.opentelemetry.io/collector/translator/conventions"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/resourcedetectionprocessor/internal"
+)
+
+const (
+	// TypeStr is the detector type string
+	TypeStr = "azure"
+)
+
+var _ internal.Detector = (*Detector)(nil)
+
+// Detector is an Azure metadata detector
+type Detector struct {
+	provider azureProvider
+}
+
+// NewDetector creates a new Azure metadata detector
+func NewDetector(component.ProcessorCreateParams, internal.DetectorConfig) (internal.Detector, error) {
+	return &Detector{provider: newProvider()}, nil
+}
+
+// Detect detects system metadata and returns a resource with the available ones
+func (d *Detector) Detect(ctx context.Context) (pdata.Resource, error) {
+	res := pdata.NewResource()
+	attrs := res.Attributes()
+
+	compute, err := d.provider.metadata(ctx)
+	if err != nil {
+		return res, fmt.Errorf("failed getting metadata: %w", err)
+	}
+
+	attrs.InsertString(conventions.AttributeCloudProvider, conventions.AttributeCloudProviderAzure)
+	attrs.InsertString(conventions.AttributeHostName, compute.Name)
+	attrs.InsertString(conventions.AttributeCloudRegion, compute.Location)
+	attrs.InsertString(conventions.AttributeHostID, compute.VMID)
+	attrs.InsertString(conventions.AttributeCloudAccount, compute.SubscriptionID)
+	attrs.InsertString("azure.vm.size", compute.VMSize)
```

Comment:
@tigrannajaryan thoughts on these? Do we have any policy about using things that aren't in semantic conventions?

Reply:
Yes, we do. See https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/common/attribute-and-label-naming.md#recommendations-for-opentelemetry-authors and https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/common/attribute-and-label-naming.md#recommendations-for-application-developers

This seems applicable:

>The name may be generally applicable to applications in the industry. In that case consider submitting a proposal to this specification to add a new name to the semantic conventions, and if necessary also to add a new namespace.

Or perhaps this:

>The name is specific to your company and may be possibly used outside the company as well. To avoid clashes with names introduced by other companies (in a distributed system that uses applications from multiple vendors) it is recommended to prefix the new name by your company's reverse domain name, e.g. com.acme.shopname

Reply:
Note that the Amazon ECS detector also uses attribute names that are not in the semantic conventions and they prefix them by `aws.ecs`.  

I think the resource group one is fairly specific to how Azure works and has no equivalents on other cloud providers so it should have some Azure prefix (if `azure` is not okay, which one should be used?) . I thought about using `host.type` instead of `azure.vm.size` but it feels like a stretch.

Reply:
You can keep it for now, but think about Azure attribute name conventions and if you can please make a proposal in spec repo.